website/docs/r/data_factory_linked_service_odata.html.markdown

```diff
@@ -0,0 +1,126 @@
+  authentication_type = "Basic"
+  url                 = "https://services.odata.org/v4/TripPinServiceRW/People"
+  username            = "emma"
+  password            = "Ch4ngeM3!"
+}
+
+resource "azurerm_data_factory_linked_service_odata" "aad_sp_key" {
+  name                                  = "example"
+  resource_group_name                   = azurerm_resource_group.example.name
+  data_factory_name                     = azurerm_data_factory.example.name
+  authentication_type                   = "AadServicePrincipal"
+  url                                   = "https://services.odata.org/v4/TripPinServiceRW/People"
+  service_principal_id                  = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+  aad_service_principal_credential_type = "ServicePrincipalKey"
+  tenant                                = "xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx"
+  aad_resource_id                       = "xxx"
+  service_principal_key                 = "xxx"
+}
+```
+
+## Argument Reference
+
+The following supported arguments are common across all Azure Data Factory Linked Services:
+
+* `name` - (Required) Specifies the name of the Data Factory Linked Service. Changing this forces a new resource to be created. Must be globally unique. See the [Microsoft documentation](https://docs.microsoft.com/en-us/azure/data-factory/naming-rules) for all restrictions.
+
+* `resource_group_name` - (Required) The name of the resource group in which to create the Data Factory Linked Service. Changing this forces a new resource
+
+* `data_factory_name` - (Required) The Data Factory name in which to associate the Linked Service with. Changing this forces a new resource.
+
+* `description` - (Optional) The description for the Data Factory Linked Service.
+
+* `integration_runtime_name` - (Optional) The integration runtime reference to associate with the Data Factory Linked Service.
+
+* `annotations` - (Optional) List of tags that can be used for describing the Data Factory Linked Service.
+
+* `parameters` - (Optional) A map of parameters to associate with the Data Factory Linfked Service.
+
+* `additional_properties` - (Optional) A map of additional properties to associate with the Data Factory Linked Service.
+
+The following supported arguments are specific to OData Linked Service:
+
+* `authentication_type` - (Required) The type of authentication used to connect to the OData source. Valid options are `AadServicePrincipal`, `Anonymous`, `Basic` and `Windows` (`ManagedServiceIdentity` are is supported at this moment by this provider)
+
+* `url` - (Required) The URL of the OData service endpoint (e.g. https://services.odata.org/v4/TripPinServiceRW/People).
+
+* `username` - (Optional) The username which can be used to authenticate to the OData endpoint.
+
+* `password` - (Optional) The password associated with the username, which can be used to authenticate to the OData endpoint.
+
+* `service_principal_id` - (Optional) The Azure Active Directory Service Principal ID.
```

Comment:
given this is only applicable to azuread should this be
```suggestion
* `azuread_service_principal_id` - (Optional) The Azure Active Directory Service Principal ID.
```

alternatively we could drop aad from the blow property, or put them in an azuread block?

Reply:
Service Principal are always in Azure AD
Looking at the code `principal_id` is always declared this way in other files of the project
I'd rather keep coherence within the project